### PR TITLE
Add comments to parser.rs

### PR DESCRIPTION
**Pull request by v-jkegler**
Add comments to parser.rs

### CODE REVIEW

File: parser/src/earley/parser.rs [hunk 1]
```diff
@@ -246,11 +246,14 @@ impl Scratch {
         }
     }
 
+    // Set current working Earley set to the single item
```
Comment (mmoskal) [inline parser/src/earley/parser.rs:249]:
```suggestion
    // Set current working Earley to empty set
```

File: parser/src/earley/parser.rs [hunk 2]
```diff
@@ -246,11 +246,14 @@ impl Scratch {
...
+    // at 'pos' in the Earley table.
```
Comment (mmoskal) [inline parser/src/earley/parser.rs:250]:
```suggestion
    // The set backing data is at `pos`
```

**Comment by mmoskal** [comment]
I did a small fix - new_row() is empty

**Comment by v-jkegler** [comment]
Good you caught this.  And it makes sense that the code should work that way.  Thanks!
________________________________

**Pull request merged**